src/inference/dev_api/ie_system_conf.h

```diff
@@ -132,4 +132,43 @@ INFERENCE_ENGINE_API_CPP(bool) with_cpu_x86_avx512_core_amx_bf16();
  */
 INFERENCE_ENGINE_API_CPP(bool) with_cpu_x86_avx512_core_amx();
 
+/**
+ * @enum cpu_core_type_of_processor
+ * @brief This enum contains defination of processor based on specific cpu core types.
+ * Will extend to support other CPU core type like ARM.
+ */
+typedef enum {
```

Comment:
why not enum class?

Reply:
@ilya-lavrenov The enum member will be used as index in table _proc_type_table.

Reply:
> I don't see any tests for this functionality

@ilya-lavrenov The whole threads pinning improvement based on CPU mapping table will include multiple PRs. And each PR only cover one of single function. The test case for whole threads pinning improvement will be handled in following PR.